compiler/src/main/kotlin/ast/mapper.kt

```diff
@@ -146,38 +172,37 @@ private fun visitDecl(node: ParseTree): Decl {
         is CellmataParser.Const_declContext -> ConstDecl(ctx = node, expr = visitExpr(node.expr()))
         is CellmataParser.State_declContext -> StateDecl(
             ctx = node,
-            body = (node.children // Get the body/code block
+            body = visitCodeBlock(node.children // Get the body/code block
                 .stream()
                 .filter { it is CellmataParser.Code_blockContext }
-                .findFirst().orElseThrow() as CellmataParser.Code_blockContext).children
-                .stream() // Iterate through content of body/code block
-                .filter { it !is TerminalNode } // Remove terminals
-                .filter { it !is CellmataParser.State_rgbContext } // Remove color declaration
-                .filter { it !is CellmataParser.State_identContext } // Remove name node
-                .map { visitStmt(it) }
-                .toList()
-        )
-        is CellmataParser.Neighbourhood_declContext -> throw NotImplementedError()
-        else -> throw AssertionError()
+                .findFirst().orElseThrow() as CellmataParser.Code_blockContext)
+        )
+        is CellmataParser.Neighbourhood_declContext -> NeighbourhoodDecl(
+            ctx = node,
+            coords = node.neighbourhood_code().coords_decl().map { Coordinate(ctx = it) }
+        )
+        is CellmataParser.Func_declContext -> FuncDecl(
+            ctx = node,
+            body = visitCodeBlock(node.code_block())
+        )
+        else -> throw AssertionError("Unexpected tree node")
     }
 }
 
+fun visitCodeBlock(block: CellmataParser.Code_blockContext): List<Stmt> {
+    return block.children
+        .filter { it !is TerminalNode } // Remove terminals
+        .filter { !(it is CellmataParser.StmtContext && it.childCount == 0) }
+        .filter { !(it is CellmataParser.StmtContext && it.getChild(0) is TerminalNode ) }
+        .map { visitStmt(it) }
+}
```

Comment:
This is weird. What is this filtering for? Why isn't children just 0 or more statements?